src/Microsoft.NetCore.Analyzers/UnitTests/Runtime/DoNotUseEnumerableMethodsOnIndexableCollectionsInsteadUseTheCollectionDirectlyTests.Fixer.cs

```diff
@@ -29,5 +30,198 @@ protected override CodeFixProvider GetCSharpCodeFixProvider()
+        {
+            VerifyCSharpFix(@"
+using System;
+using System.Collections.Generic;
+#pragma warning disable CS8019 //Unnecessary using directive
+using System.Linq;
+#pragma warning restore CS8019
+class C
+{
+    void M()
+    {
+        var list = GetList();
+        var matrix = new[] { list };
+        var f1 = Enumerable.First(list);
+        var f2 = Enumerable.First(GetList());
+        var f3 = Enumerable.First(matrix[0]);
+        Console.WriteLine(Enumerable.First(list));
+    }
+
+    IReadOnlyList<int> GetList()
+    {
+        return new List<int> { 1, 2, 3 };
+    }
+}
+", @"
+using System;
+using System.Collections.Generic;
+#pragma warning disable CS8019 //Unnecessary using directive
+using System.Linq;
+#pragma warning restore CS8019
+class C
+{
+    void M()
+    {
+        var list = GetList();
+        var matrix = new[] { list };
+        var f1 = list[0];
+        var f2 = GetList()[0];
+        var f3 = matrix[0][0];
+        Console.WriteLine(list[0]);
+    }
+
+    IReadOnlyList<int> GetList()
+    {
+        return new List<int> { 1, 2, 3 };
+    }
+}
+");
+        }
+
+        [Fact]
+        public void CA1826FixEnumerableFirstMethodChainCallWrongFormattingCSharp()
+        {
+            //this unit test documents a problematic edge case which needs to be discussed and addressed
+
+            VerifyCSharpFix(@"
+using System.Collections.Generic;
+#pragma warning disable CS8019 //Unnecessary using directive
+using System.Linq;
+#pragma warning restore CS8019
+class C
+{
+    void M()
+    {
+        var f = GetList()
+            .First();
+
+        var s = GetList()
+            .First()
+            .ToString();
+    }
+
+    IReadOnlyList<int> GetList()
+    {
+        return new List<int> { 1, 2, 3 };
+    }
+}
+", @"
+using System.Collections.Generic;
+#pragma warning disable CS8019 //Unnecessary using directive
+using System.Linq;
+#pragma warning restore CS8019
+class C
+{
+    void M()
+    {
+        var f = GetList()[0]
+;
+
+        var s = GetList()[0]
+            .ToString();
+    }
+
+    IReadOnlyList<int> GetList()
+    {
+        return new List<int> { 1, 2, 3 };
+    }
+}
+");
+        }
+
+        [Fact]
+        public void CA1826FixEnumerableFirstProblematicSyntaxCallCSharp()
+        {
+            //this unit test documents a problematic edge case which needs to be discussed and addressed
+
+            VerifyCSharpFix(@"
+using System.Linq;
+using System.Collections.Generic;
+class C
+{
+    void M()
+    {
+        GetList().First();
+    }
+
+    IReadOnlyList<int> GetList()
+    {
+        return new List<int> { 1, 2, 3 };
+    }
+}
+", @"
+using System.Linq;
+using System.Collections.Generic;
+class C
+{
+    void M()
+    {
+        GetList()[0];
+    }
+
+    IReadOnlyList<int> GetList()
+    {
+        return new List<int> { 1, 2, 3 };
+    }
+}
+", allowNewCompilerDiagnostics: true, validationMode: TestValidationMode.AllowCompileErrors);
```

Comment:
I am fine with this edge case breaking user code, as the original code seems to have a redundant `.First()` invocation. Can you please add a comment about the compiler error generated in the fixed code?